### PR TITLE
[SDK]flow as func POC

### PR DESCRIPTION
**Pull request by D-W-**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

This PR implements flow-as-func to support simple serve scenario, reference tutorials/flow-deploy/create-service-with-flow for sample usage.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_sdk/entities/_flow.py [hunk 1]
```diff
@@ -190,3 +193,17 @@ def outputs(self):
...
+        # run flow test here
+        # TODO: cache the submitter?
+        from promptflow import PFClient
+        from promptflow._sdk.operations._test_submitter import TestSubmitter
+
+        if args:
+            raise UserErrorException("Flow can only be called with keyword arguments.")
+        client = PFClient()
+        # TODO: gap: variants, environment_variables
+        result = client.flows._test_flow(flow=self, inputs=kwargs, variant=self.variant)
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/entities/_flow.py:207]: I think we should leverage FlowInvoker not directly use flow test, as we don't need stuff like the output data persist logic

Comment (D-W-) [reply src/promptflow/promptflow/_sdk/entities/_flow.py:207]: Offline synced, we need to decide whether to put connection/variant resolve logic in test submitter or in flow invoker.

File: examples/tutorials/flow-as-a-function/sample.ipynb [hunk 2]
```diff
@@ -0,0 +1,104 @@
+{
+ "cells": [
```
Comment (wangchao1230) [inline examples/tutorials/flow-as-a-function/sample.ipynb:2]: we also need a score.py example

File: src/promptflow/promptflow/_sdk/_load_functions.py [hunk 3]
```diff
@@ -62,9 +62,19 @@ def load_common(
...
     source: Union[str, PathLike, IO[AnyStr]],
+    connections=None,
+    variant=None,
     **kwargs,
 ):
-    return ProtectedFlow.load(source, **kwargs)
+    """Load a flow from source file. The loaded flow can be used as a function.
+
+    :param source: Source file path.
+    :param connections: Connections for the run.
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/_load_functions.py:72]: consider the allow streaming config

Comment (D-W-) [reply src/promptflow/promptflow/_sdk/_load_functions.py]: Supported in latest change

File: examples/tutorials/flow-as-a-function/sample.ipynb [hunk 4]
```diff
@@ -78,6 +78,50 @@
...
+   "outputs": [],
+   "source": [
+    "from promptflow import load_flow\n",
+    "from promptflow.entities import Flow\n",
+    "\n",
+    "# load the flow\n",
+    "f: Flow = load_flow(\n",
+    "   source='xx.yaml', \n",
+    "   # user can specify variant here to load a specific variant\n",
+    "   variant=\"${summarize_text_content.variant_0}\",\n",
```
Comment (wangchao1230) [inline examples/tutorials/flow-as-a-function/sample.ipynb:95]: I think we don't support variant here?

File: src/promptflow/promptflow/_sdk/_load_functions.py [hunk 5]
```diff
@@ -62,19 +62,18 @@ def load_common(
 
 def load_flow(
     source: Union[str, PathLike, IO[AnyStr]],
-    connections=None,
-    variant=None,
-    **kwargs,
 ):
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/_load_functions.py:65]: add return annotation here?

File: src/promptflow/promptflow/_sdk/_load_functions.py [hunk 6]
```diff
@@ -62,19 +62,18 @@ def load_common(
...
-    """Load a flow from source file. The loaded flow can be used as a function.
+    """Load flow from YAML file.
 
-    :param source: Source file path.
-    :param connections: Connections for the run.
-    :param variant: Variant of the run.
-    :param kwargs:
-    :return:
+    :param source: The local yaml source of a flow. Must be either a path to a local file.
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/_load_functions.py:68]: Must be either a path to a local file **or ...?**

File: examples/tutorials/flow-as-a-function/sample.ipynb (hunk 2 above)
Comment (D-W-) [reply examples/tutorials/flow-as-a-function/sample.ipynb:2]: Added a score.py example

File: examples/tutorials/flow-as-a-function/sample.ipynb (hunk 4 above)
Comment (D-W-) [reply examples/tutorials/flow-as-a-function/sample.ipynb:95]: Good catch! Forgot to update here

File: src/promptflow/promptflow/_sdk/_load_functions.py (hunk 5 above)
Comment (D-W-) [reply src/promptflow/promptflow/_sdk/_load_functions.py:65]: Shall we expose Flow entity?

File: examples/tutorials/flow-as-a-function/sample.ipynb (hunk 4 above)
Comment (D-W-) [reply examples/tutorials/flow-as-a-function/sample.ipynb]: Seems some of my local changes lost, updated.

File: src/promptflow/promptflow/_sdk/_load_functions.py (hunk 5 above)
Comment (D-W-) [reply src/promptflow/promptflow/_sdk/_load_functions.py]: Offline synced, we'll keep it private for now and add in annotation.

File: examples/tutorials/flow-deploy/create-service-with-flow/simple_score.py [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+
+@app.route("/health", methods=["GET"])
+def health():
+    """Check if the runtime is alive."""
+    return {"status": "Healthy"}
+
+
+@app.route("/score", methods=["POST"])
+def score():
```
Comment (wangchao1230) [inline examples/tutorials/flow-deploy/create-service-with-flow/simple_score.py:43]: may create a ci here

**Review by wangchao1230 — APPROVED** [review]

**Review by wangchao1230 — DISMISSED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Review by zhengfeiwang — APPROVED** [review]

File: examples/tutorials/flow-deploy/create-service-with-flow/simple_score.py (hunk 7 above)
Comment (D-W-) [reply examples/tutorials/flow-deploy/create-service-with-flow/simple_score.py:43]: Will update in next pr

**Pull request merged**